basic_module/distribution/src/impls.rs

```diff
@@ -0,0 +1,40 @@
+use crate::core::{Distribution, FeeManager};
+use crate::{Address, Deposit, StakeQuantity};
+use ckey::Ed25519Public as Public;
+use std::collections::{BTreeMap, HashSet};
+
+#[allow(dead_code)]
+struct DistributionState {
+    reward: IntermediateRewards,
+
+    // `validators` is used for inactive validators after distributing rewards
+    inactive_validators: HashSet<Address>,
+
+    // `next_validator` is used to store updated validator set to TopLevelState
+    next_validator: Vec<Validator>,
+}
+
+#[allow(dead_code)]
+struct IntermediateRewards {
+    current_reward: BTreeMap<Address, u64>,
+    calculated_reward: BTreeMap<Address, u64>,
+}
+
+#[allow(dead_code)]
+struct Validator {
+    weight: StakeQuantity,
+    delegation: StakeQuantity,
+    deposit: Deposit,
+    pubkey: Public,
+}
+
+impl Distribution for DistributionState {
+    fn distribute(&self) {}
+}
+
+#[allow(dead_code)]
+struct Manager {}
+
+impl FeeManager for Manager {
+    fn record_fee(&self, _fee: u64) {}
```

Comment:
What is the exact role of `record_fee`?
If you want to provide an interface to calculate each block fee by accumulation, I think the name should explicitly describe the behavior. Maybe something like `add_block_fee` or `accumulate_block_fee` work, but I'm not quite sure.

Also, I think you need hooks for `open_block` and `close_block`. (Maybe `close_block` can be merged into `distribute`)

Reply:
`accumulate_block_fee` looks good to me.